File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/UpdateMktDepthL2Data.java
package com.seagosoft.tws.data;

import com.ib.client.Decimal;

public class UpdateMktDepthL2Data {

    public int tickerId;
    public int position;
    public String marketMaker;
    public int operation;
    public int side;
    public double price;
    public Decimal size;
    public boolean isSmartDepth;

    public UpdateMktDepthL2Data(int tickerId, int position, String marketMaker,
                                int operation, int side, double price, Decimal size,
                                boolean isSmartDepth) {
        this.tickerId = tickerId;
        this.position = position;
        this.marketMaker = marketMaker;
        this.operation = operation;
        this.side = side;
        this.price = price;
        this.size = size;
        this.isSmartDepth = isSmartDepth;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/UpdateMktDepthL2Data.kt
package com.seagosoft.tws.wrapper.structs

import com.ib.client.Decimal


class UpdateMktDepthL2Data(
    var tickerId: Int, var position: Int, var marketMaker: String,
    var operation: Int, var side: Int, var price: Double, var size: Decimal,
    var isSmartDepth: Boolean
)